Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/FormattedError.kt

```diff
@@ -0,0 +1,13 @@
+package de.rki.coronawarnapp.util
+
+import android.content.Context
+
+interface FormattedError {
+    fun getFormattedError(context: Context): String
+}
+
+fun Throwable.tryFormattedError(context: Context): String = when {
+    this is FormattedError -> this.getFormattedError(context)
+    localizedMessage != null -> localizedMessage!!
```

Comment:
style: copy value of localizedMessage in local val, because localizedMessage could theoretically change between the two accesses

Reply:
Good point!